pkg/cloudprovider/aws/amifamily/bootstrap/eksbootstrap.go

```diff
@@ -77,3 +92,10 @@ func (e EKS) nodeLabelArg() string {
 	}
 	return fmt.Sprintf("%s%s", nodeLabelArg, strings.Join(labelStrings, ","))
 }
+
+func isIPv6(ip net.IP) bool {
+	if ip != nil {
+		return len(ip.To4()) != net.IPv4len && len(ip.To16()) == net.IPv6len
+	}
+	return false
```

Comment:
It's a trap!  😱

You can't check length, since golang's `net.IP` will often encode IPv4 addresses embedded in IPv6.  ([Example](https://go.dev/play/p/hSoy0H9mLvL))

```suggestion
	return ip.To4() == nil
```